DataScience/Experimentation.py

```diff
@@ -119,70 +120,103 @@ def detect_namespaces(j_obj, ns_set, marginal_set=None):
 
     return prop_type
 
+# Ensures min <= max and min > 0
+def check_min_max(val):
+    [val_min, val_max] = sorted([float(i) for i in val.split(',')])
+    if (val_min <= 0) or (val_min > val_max):
+        raise argparse.ArgumentTypeError("{0} is an invalid range (make sure both values are positive and min <= max".format(val))
+    return val
+
 
 if __name__ == '__main__':
 
     parser = argparse.ArgumentParser()
-    parser.add_argument('-f','--file_path', help="data file path", required=True)
-    parser.add_argument('-q','--max_q_terms', type=int, help="number of quadratic terms to explore with brute-force", default=3)
-    parser.add_argument('-p','--n_proc', type=int, help="number of parallel processors used", default=multiprocessing.cpu_count()-1)
-    parser.add_argument('-b','--base_command', help="base command before data file path (default: vw --cb_adf --dsjson -c -d )", default='vw --cb_adf --dsjson -c -d ')
+    parser.add_argument('-f','--file_path', help="data file", required=True)
+    parser.add_argument('-q','--max_q_terms', type=int, help="number of quadratic terms to explore with brute-force (default: 2)", default=2)
+    parser.add_argument('-p','--n_proc', type=int, help="number of parallel processes to use (default: auto-detect)", default=multiprocessing.cpu_count()-1)
+    # Construct default VW path in platform-agnostic away
+    vwPath = os.path.join('.', 'vw')
```

Comment:
This does not work in windows.

.\vw --cb_adf --dsjson -c -d C:\ds_workshop\ds_data_tag.json -l 0.5

FileNotFoundError: [WinError 2] The system cannot find the file specified

vw is not found

Reply:
I'll test this gain on linux and cygwin on Windows -- is there a non-cygwin way you are running vw's executable on Windows? Also, I just realized we don't currently allow the vw path to be specified via the CLI -- I'll add support for that.

Reply:
VW is on my path, and I can run from cmd: `vw --dsjson --cb_adf -d filepath.json.gz`
When I'm running in cmd: `python Experimentation.py ...` it crushes.